torch/lib/THD/base/init_methods/InitMethodFile.cpp

```diff
@@ -150,35 +150,23 @@ parseFile(std::fstream& file, rank_type world_size, std::string group_name) {
   return std::make_tuple(master_port, master_addrs, ranks);
 }
 
-rank_type getRank(const std::vector<int>& ranks, int assigned_rank,
-                  std::size_t order) {
-  if (assigned_rank >= 0) {
-    return assigned_rank;
-  } else {
-    std::vector<bool> taken_ranks(ranks.size());
-    for (auto rank : ranks) {
-      if (rank >= 0)
-        taken_ranks[rank] = true;
-    }
-
-    auto unassigned = std::count(ranks.begin(), ranks.begin() + order, -1) + 1;
-    rank_type rank = 0;
-    while (true) {
-      if (!taken_ranks[rank]) unassigned--;
-      if (unassigned == 0) break;
-      rank++;
-    }
-
-    return rank;
-  }
-}
-
 } // anonymous namespace
 
 
+InitMethod::Config initFile(std::string argument,
+                            int world_size_r,
+                            std::string group_name,
+                            int assigned_rank) {
+
+  group_name.append("#"); // To make sure it's not empty
+  std::string file_path = argument.substr(7); // chop "file://"
+  rank_type world_size;
+  try {
+    world_size = convertToRank(world_size_r);
+  } catch(std::exception& e) {
+    throw std::invalid_argument("invalid world_size");
```

Comment:
This used to give a much nicer error message before if `world_size_r == -1`, which means it was unspecified ("world_size needs to be set ..."). Can you please fix this?

Reply:
Oh, the previous code has a bug then, it was checking rank instead of world_size_r. I added it back